eng/templates/stages/mirror.yml

```diff
@@ -18,14 +18,14 @@ parameters:
 
 stages:
 - stage: Mirror
-  displayName: Mirror to DSP
+  displayName: Mirror tags & packages
```

Comment:
I realize this is literally mirroring tags but I think this would be more understandable at the stage and job level to use `source` instead of `tags`